### ISSUE TITLE
Symbol not found in flat namespace '_CVode' on new Macs with M1 chip

### ISSUE THREAD

**Issue by MichaelClerx** [myokit/myokit#861]
Two reports of this, both on Python 3.9 on MacOS 12 "Monterey", on machines with the new M1 processor

> Traceback (most recent call last):
>  File "/Users/username/Documents/pints/chi/chi/tests/test_problems.py", line 53, in setUpClass
>    cls.pd_model = chi.PharmacodynamicModel(path)
>  File "/Users/username/Documents/pints/chi/chi/_mechanistic_models.py", line 450, in __init__
>    super(PharmacodynamicModel, self).__init__(sbml_file)
>  File "/Users/username/Documents/pints/chi/chi/_mechanistic_models.py", line 44, in __init__
>    self.simulator = myokit.Simulation(self._model)
>  File "/Users/username/Documents/pints/chi/venv/lib/python3.9/site-packages/myokit/_sim/cvodessim.py", line 152, in __init__
>    self._create_simulation(cmodel.code)
>  File "/Users/username/Documents/pints/chi/venv/lib/python3.9/site-packages/myokit/_sim/cvodessim.py", line 214, in _create_simulation
>    self._sim = self._compile(module_name, fname, args, libs, libd, incd)
>  File "/Users/username/Documents/pints/chi/venv/lib/python3.9/site-packages/myokit/_sim/__init__.py", line 214, in _compile
>    return load_module(name, d_build)
>  File "/Users/username/Documents/pints/chi/venv/lib/python3.9/site-packages/myokit/_sim/__init__.py", line 56, in load_module
>    module = importlib.util.module_from_spec(spec)
>  File "<frozen importlib._bootstrap>", line 565, in module_from_spec
>  File "<frozen importlib._bootstrap_external>", line 1173, in create_module
>  File "<frozen importlib._bootstrap>", line 228, in _call_with_frames_removed
> ImportError: dlopen(/var/folders/s7/12ymf9p54jqfjjmvrv0q0z880000gn/T/tmpj8qtbzv3myokit/build/myokit_sim_21_2083898106600602248.cpython-39-darwin.so, 0x0002): symbol not found in flat namespace '_CVode'

In one case it just went away again...

**Comment by MichaelClerx** [comment]
Few similar issues in other projects:

- https://stackoverflow.com/questions/69973873/symbol-not-found-in-flat-namespace-ft-done-face-from-reportlab-with-python3
- https://github.com/Homebrew/discussions/discussions/2693
- https://github.com/rofl0r/proxychains-ng/issues/409

**Comment by MichaelClerx** [comment]
My best guess for this is that binaries for the wrong architecture are being installed by something, somewhere (e.g. sundials binaries by homebrew?)

**Comment by MichaelClerx** [comment]
>  (e.g. sundials binaries by homebrew?)

Also that bit seems to be working, for some people? 
https://apple.stackexchange.com/questions/437569/error-after-installing-brew-formula-in-arm64-architecture

**Comment by mirams** [comment]
I had the same thing on MacOS Ventura, and cured it by putting 
```
[sundials]
# Location of sundials shared libary files (.so, .dll, or .dylib).
# Multiple paths can be set using ; as separator.
lib = /opt/homebrew/lib
# Location of sundials header files (.h).
inc = /opt/homebrew/include
```

**Comment by MichaelClerx** [comment]
https://www.reddit.com/r/MacOS/comments/jw9guu/why_did_homebrew_move_from_usrlocalto_opthomebrew/

**Comment by MichaelClerx** [comment]
> My best guess for this is that binaries for the wrong architecture are being installed by something, somewhere (e.g. sundials binaries by homebrew?)

Adding opt/homebrew should fix this.

Unable to reproduce this issue exactly on an M1, although very similar issue seen. Assuming it's the same and closing!

**Comment by MichaelClerx** [comment]
Thanks @mirams !

**Comment by MichaelClerx** [comment]
Got another report of this error on an M1 chip with 1.33.9 and a freshly generated myokit.ini

**Comment by MichaelClerx** [comment]
Hey @martinjrobins , do you have any ideas about this one?

**Comment by martinjrobins** [comment]
sorry, I've not got an M1 to try this on, looks like `brew install sundials` is meant to work ok on M1's https://doesitarm.com/formula/sundials/

**Comment by mirams** [comment]
I think you need to ask exactly what commands were typed into where, whether they had installed xcode and stuff, and what exactly the error message is. I had to type `python3`/`pip3` rather than `python`/`pip` and I had to do the `brew install pyqt5` because the pip one failed for some reason.

But `brew install sundials` definitely worked.

**Comment by MichaelClerx** [comment]
Thanks both!

@mirams how did you install Python?

Frankie has an M1 and doesn't have the "flat namespace" issue, he installed Python
> via the installer package from the Python website

I think it might be related to slightly different ways of compiling stuff used by conda and the official distro, although I really don't understand how that affects myokit and cvode (maybe via `setuptools`)

**Comment by MichaelClerx** [comment]
https://stackoverflow.com/questions/35006614/what-does-symbol-not-found-expected-in-flat-namespace-actually-mean

**Comment by MichaelClerx** [comment]
https://docs.brew.sh/C++-Standard-Libraries

**Comment by MichaelClerx** [comment]
Confirmation in that one "flat namespace" issue was on a conda install

**Comment by MichaelClerx** [comment]
Confirmed that replacing with brew version solves problem

**Comment by MichaelClerx** [comment]
Will be solved by #864
